sesman/config.c

```diff
@@ -279,9 +280,20 @@ config_read_security(int file, struct config_security *sc,
                     unrecognised);
             }
         }
+        if (0 == g_strcasecmp(buf, SESMAN_CFG_SEC_ALLOWED_ALTERNATE_SHELLS))
+        {
```

Comment:
Need `g_free(sc->allowed_alternate_shells);` here.

You need this regardless of whether the default is set earlier or not, as the user might include this multiple times by error, thus resulting in a memory leak.